src/resources/sgmanager.c

```diff
@@ -143,12 +131,28 @@ int sg_load(sd_chr_file *chr, const char *pilotname) {
 }
 
 int sg_save(sd_chr_file *chr) {
-    char filename[1024];
+    char target_filename[1024];
+    char scan_filename[1024];
+
     const char *dirname = pm_get_local_path(SAVE_PATH);
-    snprintf(filename, sizeof(filename), "%s%s.CHR", dirname, chr->pilot.name);
-    int ret = sd_chr_save(chr, filename);
+    snprintf(target_filename, sizeof(target_filename), "%s%s.CHR", dirname, chr->pilot.name);
+    strncpy_or_abort(scan_filename, target_filename, sizeof(scan_filename));
+
+    // Try to make the save uppercase.
```

Comment:
This comment does not relate to the call to `scan_directory_for_file` and `rename` below.

The transformation we should be applying to get the savegame name (as per the original game):
- Any character a through z or A through Z, replace with a capital letter A-Z.
- Any other character, replace with an underscore `_`
- (Append .CHR)

As an example, my character whose name is `jo dU-lI` gets saved to filename `JO_DU_LI.CHR` in OMF 2097.

As for the scan_directory_for_file and rename call below.. Why do we need a case insensitive scan? We know the old (oomf 0.8.0 and prior) filename, it's just `name .. ".CHR"`. and we know the new filename, as described above. Check if the file exists with `fopen` (standard C) or `stat` (POSIX) before deciding to rename.